sw/device/tests/dif/dif_rv_timer_sanitytest.c

```diff
@@ -77,19 +77,29 @@ bool test_main(void) {
   uint64_t current_time;
   CHECK(dif_rv_timer_counter_read(&timer, kHart, &current_time) ==
         kDifRvTimerOk);
+
+  uint64_t timer_threshold = current_time + kDeadline;
   LOG_INFO("Current time: %d; timer theshold: %d", (uint32_t)current_time,
-           (uint32_t)(current_time + kDeadline));
+           (uint32_t)(timer_threshold));
   CHECK(dif_rv_timer_arm(&timer, kHart, kComparator,
                          current_time + kDeadline) == kDifRvTimerOk);
 
   irq_fired = false;
   CHECK(dif_rv_timer_counter_set_enabled(&timer, kHart, kDifRvTimerEnabled) ==
         kDifRvTimerOk);
 
-  while (!irq_fired) {
-    LOG_INFO("Waiting...");
-    wait_for_interrupt();
-  }
+  LOG_INFO("Waiting...");
+  wait_for_interrupt();
+  CHECK(irq_fired);
+
+  CHECK(dif_rv_timer_counter_read(&timer, kHart, &current_time) ==
+        kDifRvTimerOk);
+  LOG_INFO("Current time: %d", (uint32_t)current_time);
+
+  CHECK(current_time >= timer_threshold,
+        "Current time expected to exceed the set deadline");
+  CHECK(current_time <= (timer_threshold + 10),
+        "Current time expected to not exceed the set deadline + 10");
```

Comment:
Why 10? This seems flaky, since nothing actually stops the timer when the IRQ is fired. Logging, in particular, takes a long time on non-DV platforms. If you really want to keep this, you should stop the timer in the IRQ handler, and include a comment specifying that it's there so that this deadline can  be checked.

Reply:
The ISR does disable the timer, but the ISR also has logs, so it might not cut it. This is not required anyway (already verified at IP level DV). Removed.